thoth/storages/sync.py

```diff
@@ -598,8 +598,9 @@ def sync_documents(
 
     stats = dict.fromkeys(HANDLERS_MAPPING, (0, 0, 0, 0))
     for handler, documents in (handlers or static_handlers).items():
-        stats[handler] = HANDLERS_MAPPING[handler](
-            documents, force=force, graceful=graceful, graph=graph, is_local=is_local
-        )
+        if documents is None or documents:
```

Comment:
How could `documents` be `None` if it is initialized above to `[]`?

And even in that case, shouldn't that be:

```suggestion
        if documents:
```

Reply:
The function sync_documents default for var `document_ids` is None, if no value is provided for `document_ids` https://github.com/thoth-station/storages/blob/f8d5406d45811ce38bd703f3726614cd571a7553/thoth/storages/sync.py#L571
so , when its `document_ids` is None, it would fall into the else condition, where static_handlers are set for value None
https://github.com/thoth-station/storages/blob/f8d5406d45811ce38bd703f3726614cd571a7553/thoth/storages/sync.py#L596
in this case, when for loop over static_handler
https://github.com/thoth-station/storages/blob/f8d5406d45811ce38bd703f3726614cd571a7553/thoth/storages/sync.py#L600
the documents would be `None`

Reply:
I see, thanks.

But: do we want to try to call the handler if there are no documents is `None`?

Reply:
There are 3 values in play here: `actual document id, None, []`
- When the actual document id is present, the idea is to call only that handle(for example `sync_solver_document`), so that would only get synced, and other sync functions are not called.
- When None values, the idea is to execute all the sync functions, so they can sync the all document present ceph.

- when [] is value, it just states, one of the handlers is having document_id. so executing it not all the functions should be executed. Only the handler with document_id. 

please let me know if you feel otherwise.

Reply:
I would rather have 
```suggestion
        if documents != []:
```
Which look more obvious on why we're excluding that handler.
Or maybe we could turn handlers into a defaultdict

Reply:
Although I might have had typing incompatibilites while working on that piece of code which prevented me to do just that, can't remember exactly.

Reply:
Made the change suggested.
Ready for review again.

Reply:
+1 on the defaultdict because then there is no need to check for `[]`

Reply:
i m unable to understand the use of defaultdict in here.
defaultdict would still require a value.
and still would have to check that right?

Reply:
The point of `defaultdict` would be that we don't need to check for `[]` in `handlers`, because `handlers` member are only populated with a handler key the first time we add a document to it.

However, the code is fine by me in its current form and better is the enemy of good.